sw/device/lib/dif/dif_uart.h

```diff
@@ -0,0 +1,284 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _SW_DEVICE_LIB_DIF_UART_H_
+#define _SW_DEVICE_LIB_DIF_UART_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * UART interrupt configuration
+ *
+ * Enumeration used to enable, disable, test and query the UART interrupts.
+ * Please see the comportability specification for more information:
+ * https://docs.opentitan.org/doc/rm/comportability_specification/
+ */
+typedef enum dif_uart_interrupt {
+  kDifUartTxWatermark = 0, /**< TX FIFO dipped below watermark */
+  kDifUartRxWatermark,     /**< RX FIFO reached high watermark */
+  kDifUartTxEmpty,         /**< TX FIFO empty */
+  kDifUartRxOverflow,      /**< RX FIFO overflow */
+  kDifUartRxFrameErr,      /**< RX framing error */
+  kDifUartRxBreakErr,      /**< RX break condition */
+  kDifUartRxTimeout,       /**< RX FIFO timeout (not empty in a set time) */
+  kDifUartRxParityErr,     /**< RX parity error detection */
+} dif_uart_interrupt_t;
+
+/**
+ * TX and RX FIFO depth configuration
+ *
+ * Enumeration used to set the upper limit of bytes to queue.
+ */
+typedef enum dif_uart_watermark {
+  kDifUartWatermarkByte1 = 0, /**< 1 byte */
+  kDifUartWatermarkByte4,     /**< 4 bytes */
+  kDifUartWatermarkByte8,     /**< 8 bytes */
+  kDifUartWatermarkByte16,    /**< 16 bytes */
+  kDifUartWatermarkByte30,    /**< 30 bytes */
+} dif_uart_watermark_t;
+
+/**
+ * Generic enable/disable enumeration
+ *
+ * Enumeration used to enable/disable bits, flags, ...
+ */
+typedef enum dif_uart_enable {
+  kDifUartEnable = 0, /**< enable */
```

Comment:
Well I feel silly asking for this: `kDifUartEnableEnable`, `kDifUartEnableDisable`.  I think I'm willing to leave it as is.